Data race leading to panic

Hi!

I faced a bug that is very hard to reproduce. The log below is from an old version of `routedns`, however that part of code has not changed for a long time:

```
panic: close of closed channel

goroutine 45872 [running]:
github.com/folbricht/routedns.(*request).markDone(...)
        /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.21-0.20230911182951-20545c9a53e3/pipeline.go:225
github.com/folbricht/routedns.(*Pipeline).start.func2()
        /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.21-0.20230911182951-20545c9a53e3/pipline.go:109 +0x3ed
created by github.com/folbricht/routedns.(*Pipeline).start in goroutine 31
        /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.21-0.20230911182951-20545c9a53e3/pipeline.go:101 +0x808
```

My guess was that it has been triggered by data race and I indeed was able to to find it with race detector when sending a lot of concurrent requests:

```
==================
WARNING: DATA RACE
Write at 0x00c00005c028 by goroutine 13:
  github.com/folbricht/routedns.(*Pipeline).start()
      /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.51/pipeline.go:87 +0x1aa
  github.com/folbricht/routedns.NewPipeline.func1()
      /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.51/pipeline.go:48 +0x33

Previous read at 0x00c00005c028 by goroutine 122428:
  github.com/folbricht/routedns.(*Pipeline).start.func1()
      /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.51/pipeline.go:99 +0x33

Goroutine 13 (running) created at:
  github.com/folbricht/routedns.NewPipeline()
      /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.51/pipeline.go:48 +0x230
  github.com/folbricht/routedns.NewDNSClient()
      /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.51/dnsclient.go:59 +0x1c7
<omitted>

Goroutine 122428 (running) created at:
  github.com/folbricht/routedns.(*Pipeline).start()
      /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.51/pipeline.go:99 +0x624
  github.com/folbricht/routedns.NewPipeline.func1()
      /home/dmitry/go/pkg/mod/github.com/folbricht/routedns@v0.1.51/pipeline.go:48 +0x33
==================
Found 1 data race(s)
```

As you can see, this output is from `routedns@v0.1.51`. I also reproduced this in `v0.1.21-0.20230911182951-20545c9a53e3`.

There is a [classical mistake][loopvar] of capturing a loop variable into a [new goroutine][mistake]. The simplest fix is to switch to go 1.22 in `go.mod`. Another option is to make a copy of `req` and use it in the goroutine.

[loopvar]: https://go.dev/blog/loopvar-preview
[mistake]: https://github.com/folbricht/routedns/blob/3aa67c59d3272ee5f568420ea73f5c2c0dfee3bd/pipeline.go#L99

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/folbricht/routedns
 
-go 1.20
+go 1.22
 
 require (
 	github.com/BurntSushi/toml v1.2.1

